log/log.go

```diff
@@ -134,9 +138,56 @@ func (m *defaultLogger) LogMessage(message string, level corelog.Level) {
 	m.logMessage(message, level)
 }
 
+func (m *defaultLogger) PrintBitriseStartedEvent(plan models.WorkflowRunPlan) {
+	if m.opts.LoggerType == JSONLogger {
+		m.logger.LogEvent(plan, corelog.EventLogFields{
+			Timestamp: m.opts.TimeProvider().Format(rfc3339MicroTimeLayout),
+			EventType: "bitrise_started",
+		})
+	} else {
+		m.PrintBitriseASCIIArt()
+		m.Warnf("CI mode: %v", plan.CIMode)
+		m.Warnf("PR mode: %v", plan.PRMode)
+		m.Warnf("Debug mode: %v", plan.DebugMode)
+		m.Warnf("Secret filtering mode: %v", plan.SecretFilteringMode)
+		m.Warnf("Secret Envs filtering mode: %v", plan.SecretEnvsFilteringMode)
+		m.Warnf("No output timeout mode: %v", plan.NoOutputTimeoutMode)
+		m.Print()
+		var workflowIDs []string
+		for _, workflowPlan := range plan.ExecutionPlan {
+			workflowID := workflowPlan.WorkflowID
+			if workflowPlan.WorkflowID == plan.TargetWorkflowID {
+				workflowID = colorstring.Green(workflowPlan.WorkflowID)
+			}
+			workflowIDs = append(workflowIDs, workflowID)
+		}
+		var prefix string
+		if len(workflowIDs) == 1 {
+			prefix = colorstring.Blue("Running workflow")
+		} else {
+			prefix = colorstring.Blue("Running workflows")
+		}
+
+		m.Printf("%s: %s", prefix, strings.Join(workflowIDs, " -->  "))
+	}
+}
```

Comment:
Should not only the internal sub loggers (json and console) know how to write the events to the output and the main logger is just a wrapper around them?

So this function would be simply:
```
func (m *defaultLogger) PrintBitriseStartedEvent(plan models.WorkflowRunPlan) {
    m.logger.LogEvent(plan, corelog.EventLogFields{
        Timestamp: m.opts.TimeProvider().Format(rfc3339MicroTimeLayout), // this might be optional or maybe even the logger should know about how to format the received time
 	EventType: "bitrise_started",
    })
```
Then in the console logger we would have a new property:
```
type consoleLogger struct {
	output io.Writer
        generator messageGenerator 
}
```
and this `messageGenerator` type will be a preparation for the message conversion feature we will need to provide. Because we already know that we will need to extract the logic which converts a `WorkflowRunPlan` struct restored from the json log. 

This message generator would have multiple properties which would return an array of log level and message pairs:
```
type GeneratedLogMessage struct {
	level   corelog.Level
	message string
}

func (m *messageGenerator) bitriseStartedMessages(plan models.WorkflowRunPlan) []GeneratedLogMessage {
	...
}
```
And then the console logger would do:
```
func (l consoleLogger) LogBitriseStarted(plan models.WorkflowRunPlan) {
	generatedLogMessages := l.generator.bitriseStartedMessages(plan)

	for _, generatedLogMessage := range generatedLogMessages {
		switch generatedLogMessage.level {
		case WarnLevel: 
			l.warn(generatedLogMessage.message)
		case ErrorLevel:
			l.error(generatedLogMessage.message)
		}
		...
	}
}
```
This way when we are working on the json -> console message conversion we already have most of the logic isolated in a single place. And we only need to write a struct which generates the received json line into the corresponding struct. Then feeds this to the already existing message generator and returns the array of converted color coded lines.

Reply:
We had discussion about this and agreed to delay the decision making until the json -> console log message conversion task. We were not sure what would be the best decision here at the moment. So we will first handle all of the events and then think about how to merge these with the message conversion feature.